Source/ProgressionSystemRuntime/Public/Data/PSWorldSubsystem.h

```diff
@@ -87,18 +91,22 @@ class PROGRESSIONSYSTEMRUNTIME_API UPSWorldSubsystem : public UWorldSubsystem
 	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Progression System HUD Component"))
 	TObjectPtr<class UPSHUDComponent> PSHUDComponentInternal = nullptr;
 
-	/** Progression System Spot Component reference*/
-	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Progression System HUD Component"))
+	/** Progression System Array of Spot Components */
+	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Progression System Spot Array"))
 	TArray<class UPSSpotComponent*> PSSpotComponentArrayInternal;
 
 	/** Progression System Spot Component reference*/
-	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Progression System HUD Component"))
-	TObjectPtr<class UPSSpotComponent> PSSpotComponentInternal = nullptr;
+	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Progression System Spot Component"))
+	TObjectPtr<class UPSSpotComponent> PSCurrentSpotComponentInternal = nullptr;
```

Comment:
Please don't cache `PSCurrentSpotComponentInternal` and `PSLastSpotComponentInternal` as we don't want to pay cost on maintaining them. Instead, figure out differently without caching. LIke find needed spot by comparing Spot's PlayerTag with LocalPlayerCharacter's PlayerTag

Reply:
ProgressionSystem currently changes the player so once a GetLocalPlayerCharacter is called it returns a new character instead of old one, however the Mainmenu keeps showing the old character